kitsune/tidings/events.py

```diff
@@ -108,8 +104,30 @@ class Event(object):
     #: ``set(['color', 'flavor'])``
     filters = set()
 
-    def fire(self, exclude=None, delay=True):
-        """Notify everyone watching the event.
+    @classmethod
+    def fire_async(cls, instance=None, exclude=None):
+        """
+        Enqueues a JSON-serializer-friendly Celery task that builds and sends emails
+        to everyone watching the event, excluding any users provided by "exclude". The
+        event will be constructed with the given instance if provided.
+        """
+        kwargs = {}
+        if instance:
+            kwargs.update(
+                instance_cls_module_name=instance.__class__.__module__,
+                instance_cls_name=instance.__class__.__name__,
+                instance_id=instance.id,
+            )
+        if exclude:
+            if not isinstance(exclude, Sequence):
+                exclude = [exclude]
+            kwargs.update(exclude_user_ids=[user.id for user in exclude])
+
+        fire.delay(cls.__module__, cls.__name__, **kwargs)
```

Comment:
This is very confusing. The fact that the Event class has a method `fire` but we also have a `fire` task

Reply:
Sorry, I think that's a bad choice of mine to name the Celery task `fire` as well (too many meanings for `fire`). We can change the name of the Celery task if we stay with this approach.